### PR TITLE
fix(cli): Watch workspace members if tauri dir is workspace root

### LINKED ISSUE #8509: [bug] After tuari project dependency is installed, running `pnpm tauri dev` cannot run,
### Describe the bug

After tuari project dependency is installed, running `pnpm tauri dev` cannot run,
report: thread '<unnamed>' panicked at src\interface\rust.rs:451:79: called `Option::unwrap()` on a `None` value stack backtrace: note: Some details are omitted, run with `RUST_BACKTRACE=full` for a verbose backtrace. * The terminal will be reused by the task, press any key to close.

### Reproduction

_No response_

### Expected behavior

Can operate normally

### Full `tauri info` output

```text
thread '<unnamed>' panicked at src\interface\rust.rs:451:79: called `Option::unwrap()` on a `None` value stack backtrace: note: Some details are omitted, run with `RUST_BACKTRACE=full` for a verbose backtrace. * The terminal will be reused by the task, press any key to close.
```

### Stack trace

_No response_

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by FabianLars**
See title. This PR also includes a fix/workaround for paths with funny characters that may not make the glob expansion panic.

Fixes #8509 -> That's about the funny characters and the reason i looked into the code again in the first place.

### CODE REVIEW

**Comment by FabianLars** [comment]
thinking about it i'm not sure anymore if it actually made a difference since it should watch the workspace members anyway since they are inside the root (if tauri dir is ws root) but ig being mroe explicit about it is good? 🙃

File: tooling/cli/src/interface/rust.rs [hunk 1]
```diff
@@ -415,40 +415,43 @@ impl Rust {
...
-    let watch_folders = watch_folders
-      .into_iter()
-      .flat_map(|p| {
-        match expand_member_path(&p) {
-          Ok(p) => p,
-          Err(err) => {
-            // If this fails cargo itself should fail too. But we still try to keep going with the unexpanded path.
-            error!("Error watching {}: {}", p.display(), err.to_string());
-            vec![p]
+    // We also try to watch workspace members, no matter if the tauri cargo project is the workspace root or a workspace member
```
Comment (amrbashir) [inline tooling/cli/src/interface/rust.rs]: this logic now needs a function on its own IMO.

**Review by amrbashir — CHANGES_REQUESTED** [review]

File: tooling/cli/src/interface/rust.rs (hunk 1 above)
Comment (FabianLars) [reply tooling/cli/src/interface/rust.rs]: Just the "get workspace member paths" part or the whole "get watch folders" block? It's only a one line difference so excuse the stupid question but i don't wanna do it twice :D

Comment (amrbashir) [reply tooling/cli/src/interface/rust.rs]: hard to say tbh, just go with your gut, I am fine with either way

**Review by amrbashir — APPROVED** [review]

**Pull request merged**